pkg/broker/ingress/handler_test.go

```diff
@@ -140,52 +161,99 @@ func TestHandler(t *testing.T) {
 
 			// If event is accepted, check that it's stored in the decouple sink.
 			if res.StatusCode == nethttp.StatusOK {
+				m, err := rec.Receive(ctx)
+				if err != nil {
+					t.Fatal(err)
+				}
+				savedToSink, err := binding.ToEvent(ctx, m)
+				if err != nil {
+					t.Fatal(err)
+				}
 				// Retrieve the event from the decouple sink.
-				fakeClient := h.decouple.(*multiTopicDecoupleSink).client.(*fakePubsubClient)
-				savedToSink := <-fakeClient.topics[topic]
 				if tc.event.ID() != savedToSink.ID() {
 					t.Errorf("Event ID mismatch. got: %v, want: %v", savedToSink.ID(), tc.event.ID())
 				}
 				if savedToSink.Time().IsZero() {
 					t.Errorf("Saved event should be decorated with timestamp, got zero.")
 				}
 			}
+
+			select {
+			case <-ctx.Done():
+				t.Fatalf("test cancelled or timed out: %v", ctx.Err())
+			default:
+			}
 		})
 	}
 }
 
+func createPubsubClient(ctx context.Context, t *testing.T, psSrv *pstest.Server) (*pubsub.Client, func()) {
```

Comment:
Can we move this to a util package so other tests can use it too?

Reply:
I'll move these out once I can revert the cleanup change.